### PR TITLE
feat(addon-doc): use `anchorScrolling` by default

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the new behaviour?

Part of https://github.com/taiga-family/taiga-ui/pull/6077

### CODE REVIEW

File: projects/addon-doc/components/navigation/navigation.component.ts [hunk 1]
```diff
@@ -82,24 +81,45 @@ export class TuiDocNavigationComponent {
...
+    ngOnInit(): void {
+        combineLatest([
+            this.title$.pipe(switchMap(() => this.readyToScroll$.pipe(filter(Boolean)))),
+            this.router.events.pipe(
+                filter((event): event is Scroll => event instanceof Scroll),
+            ),
+        ])
+            .pipe(
+                take(1),
+                map(([, event]) => event),
```
Comment (waterplea) [inline projects/addon-doc/components/navigation/navigation.component.ts:109]: Let's reorder those so we do not need an empty first array element destruct.

File: projects/addon-doc/components/navigation/navigation.component.ts [hunk 2]
```diff
@@ -82,24 +81,45 @@ export class TuiDocNavigationComponent {
...
-        // https://stackoverflow.com/questions/36101756/angular2-routing-with-hashtag-to-page-anchor
         title$.subscribe(title => {
             cdr.markForCheck();
             titleService.setTitle(title);
             this.openActivePageGroup();
-            this.handleAnchorLink(this.activatedRoute.snapshot.fragment || '');
         });
     }
 
+    ngOnInit(): void {
```
Comment (waterplea) [inline projects/addon-doc/components/navigation/navigation.component.ts:100]: Why not in constructor?

File: projects/addon-doc/components/navigation/navigation.component.ts [hunk 3]
```diff
@@ -222,11 +236,11 @@ export class TuiDocNavigationComponent {
...
-        element.classList.add('tui-doc-animated-example');
-        element.scrollIntoView({
-            block: 'start',
-            inline: 'nearest',
-            behavior: this.scrollBehavior,
-        });
+        this.viewportScroller.scrollToAnchor(fragment);
+
+        element.classList.add('_target');
+        timer(1000)
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/components/navigation/navigation.component.ts:242]: `:target` does not restart the animation `tuiShaking`

File: projects/addon-doc/components/navigation/navigation.component.ts [hunk 4]
```diff
@@ -82,24 +81,45 @@ export class TuiDocNavigationComponent {
...
             cdr.markForCheck();
             titleService.setTitle(title);
             this.openActivePageGroup();
-            this.handleAnchorLink(this.activatedRoute.snapshot.fragment || '');
         });
     }
 
+    ngOnInit(): void {
+        combineLatest([
+            this.title$.pipe(switchMap(() => this.readyToScroll$.pipe(filter(Boolean)))),
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/components/navigation/navigation.component.ts:102]: we have now:
image


`just a 200ms timeout`

it's default fallback
image

File: projects/addon-doc/components/navigation/navigation.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/components/navigation/navigation.component.ts]: fixed

File: projects/demo/src/styles.less [hunk 5]
```diff
@@ -1,6 +1,10 @@
 /* You can add global styles to this file, and also import other style files */
 @sidebar-width: 15.5rem;
 
+html {
+    scroll-behavior: smooth;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/styles.less]: fixed

File: projects/addon-doc/components/navigation/navigation.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-doc/components/navigation/navigation.component.ts]: fixed

File: projects/core/styles/mixins/mixins.less [hunk 6]
```diff
@@ -428,7 +428,9 @@
 .scroll-behavior(@mode: smooth) {
     scroll-behavior: @mode;
 
-    :host-context(tui-root._reduced-motion) & {
+    &._reduced-motion,
+    ._reduced-motion > &,
+    :host-context(._reduced-motion) & {
```
Comment (waterplea) [inline projects/core/styles/mixins/mixins.less:433]: I don't like this selector, it's better not to reach outside `tui-root`. Let's roll this back and just keep smooth-scrolling always on, it's in the demo app anyway, not in the addon-doc.

File: projects/core/components/root/root.component.ts [hunk 7]
```diff
@@ -65,5 +65,9 @@ export class TuiRootComponent {
         @Inject(TUI_THEME) theme: string,
     ) {
         body.setAttribute('data-tui-theme', theme.toLowerCase());
+
+        if (reducedMotion) {
```
Comment (waterplea) [inline projects/core/components/root/root.component.ts:69]: Let's roll this back too.

File: projects/addon-doc/components/navigation/navigation.component.ts [hunk 8]
```diff
@@ -82,22 +80,34 @@ export class TuiDocNavigationComponent {
...
+
+        combineLatest([
+            this.router.events.pipe(
+                filter((event): event is Scroll => event instanceof Scroll),
+            ),
+            this.title$.pipe(switchMap(() => this.readyToScroll$.pipe(filter(Boolean)))),
+        ])
+            .pipe(
+                take(1),
+                map(([event]) => event),
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/components/navigation/navigation.component.ts]: fixed

File: projects/core/styles/mixins/mixins.less (hunk 6 above)
Comment (splincode) [reply projects/core/styles/mixins/mixins.less]: fixed

File: projects/addon-doc/components/navigation/navigation.component.ts [hunk 9]
```diff
@@ -82,22 +80,34 @@ export class TuiDocNavigationComponent {
...
+            ),
+            this.title$.pipe(switchMap(() => this.readyToScroll$.pipe(filter(Boolean)))),
+        ])
+            .pipe(
+                take(1),
+                map(([event]) => event.anchor),
+                filter(Boolean),
+                takeUntil(this.destroy$),
+            )
+            .subscribe(anchor => this.navigateToAnchorLink(anchor as string));
```
Comment (waterplea) [inline projects/addon-doc/components/navigation/navigation.component.ts:110]: Why do we need the typecast?

**Review by waterplea — APPROVED** [review]

File: projects/addon-doc/components/navigation/navigation.component.ts (hunk 9 above)
Comment (splincode) [reply projects/addon-doc/components/navigation/navigation.component.ts:110]: image

File: projects/demo/src/modules/app/app.routes.ts [hunk 10]
```diff
@@ -1972,8 +1972,9 @@ export const ROUTES: Routes = [
 @NgModule({
     imports: [
         RouterModule.forRoot(ROUTES, {
-            initialNavigation: `enabledBlocking`,
             scrollPositionRestoration: `enabled`,
+            anchorScrolling: `enabled`,
+            scrollOffset: [0, 64],
```
Comment (nsbarsukov) [inline projects/demo/src/modules/app/app.routes.ts:1977]: Should we drop this property ? 🤔 

https://github.com/taiga-family/taiga-ui/blob/3ef4f4892ff6a01333ca8a185bd8de283c675d62/projects/addon-doc/components/main/main.style.less#L6

Comment (splincode) [reply projects/demo/src/modules/app/app.routes.ts:1977]: 64px === 4rem

**Pull request merged**